entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/extension/KotlinTypeExt.kt

```diff
@@ -52,3 +62,71 @@ private val coreTypes = listOf(
     "Vector2",
     "Vector3"
 )
+
+/**
+ * taken from the "inherited by" section here: https://docs.godotengine.org/en/stable/classes/class_resource.html
+ */
+private val resourceTypes = listOf(
+    "godot.Animation",
```

Comment:
I don't think we need to list all resource types, a check that a type inherits from `godot.Resource` should be enough. This also means that when new resource types are added, we don't have to update this list, it just works out of the box.

Reply:
That's not possible as the exact type is needed for the editor to show the correct subInspector: https://github.com/godotengine/godot/blob/5f1107aa30295e686be6f41cb6d17fc2cff1e036/editor/editor_properties.cpp#L3287

Reply:
I don't get it, we just use this to check if it's a resource right? Can we just check if the class inherits from `godot.Resource`?

Reply:
No.
This one checks if it's a resource at all:https://github.com/utopia-rise/godot-kotlin/blob/1a46bc80655f157f1fb998de0eb7452e18047eb7/entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/extension/KotlinTypeExt.kt#L12

This one checks if it's a registrable resource (if it does not derive from a godot resource, it IS a godot resource): https://github.com/utopia-rise/godot-kotlin/blob/1a46bc80655f157f1fb998de0eb7452e18047eb7/entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/extension/KotlinTypeExt.kt#L20

And this one gets the type of the resource we use for the typehint string: https://github.com/utopia-rise/godot-kotlin/blob/1a46bc80655f157f1fb998de0eb7452e18047eb7/entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/extension/KotlinTypeExt.kt#L24

And we need the godot resource type to be used in the hint so the editor knows which subinspector to show. If we don't provide it, it just shows the default subInspector for resources and not the individual once for textures, colors and so on.

Reply:
We can still provide that value without that list right? i.e just get a simple name from fqname of the class?

Reply:
On `fun KotlinType.isAnyResourceType()` the fact that it inherits from `Resource` then it should be registerable. Are we protecting against users declaring their own `Resource` types the engine does not support yet? My concern is that we have to maintain that list, when say there's a new resource type added by Godot.

Reply:
Yes with this impl we prevent that. And i think we should. Otherwise users have the problem @piiertho had this weekend with trying to register custom resources and nothing works the way it should.
So yes i get your point with the additional maintaining overhead, but we already have that in this file for all the core types.

Reply:
@chippmann the problem I had this weekend was a godot-cpp bug. Custom resources default values are not well initialized in godot-cpp. But it seems to work in D bindings. So it seems that GDNative allow user to define their own resource types.

Reply:
So we should allow it then and look at what the D-Binding has done?
And i think you are confusing something. Default values are set through code. What is set in the editor are not default values. And IIRC the editor does not support default values for resources at all.
What you probably mean is the cpp-binding has a bug with setting values that were defined in the editor (!= default values).

Reply:
I've been scouring the godot issues page and it looks like you can define custom resource? https://github.com/godotengine/godot/issues/27470 as an example. I guess they would inherit from `godot.Resource` directly as they would probably want to get reference counting support?

Reply:
In godot cpp you have to access it through `Ref<CustomResource>` in oreder to have reference counting and avoid dangling pointers.
@chippmann yes I'm talking about default values in code, that's buggy in cpp bindings.

Reply:
Hmmm, I think what we have now should suffice and lets see what users will say.

Reply:
I agree
But back to the origins of this discussion; should we now prevent the registration of custom resources for now (which requires this list) or shouldn't we prevent it and possibly let user experience undefined behavior but less to maintain for us (remove this list)?

Reply:
@chippmann I think we should not prevent registration of custom resources, as other bindings does not do that